Find_first_missing_positive_integer.rb

```diff
@@ -0,0 +1,34 @@
+=begin 
+This problem was asked by Stripe.
+
+Given an array of integers, find the first missing positive integer in linear time and constant space. 
+In other words, find the lowest positive integer that does not exist in the array. 
+The array can contain duplicates and negative numbers as well.
+
+For example, the input [3, 4, -1, 1] should give 2. The input [1, 2, 0] should give 3.
+
+You can modify the input array in-place.
+
+=end 
+
+def findfirstmissingpostiveint(array)
+    array = array.sort 
+    array.each_with_index do |number, index|
+        answer = 0 

```

Comment:
what if the missing number is 0. so [-2, -4, -3, -1, 1, 2]

Reply:
The input [1, 2, 0] should give 3. Because of this example i had to build code like this.

Reply:
input [3, 4, -1, 1] should give 2.  this too.

Reply:
Technically 0 should come which actually comes. but i had to hard code to move the next number

Reply:
if i remove the condition (answer != 0 ) i can get the correct missing value for all the question except for one " [3, 4, -1, 1] should give 2" but this yeah  [-2, -4, -3, -1, 1, 2] is 0 i can get it.  I dont know which one is needed according to the question the condition is very much needed, else we can avoid the condition and get  the correct and relevant answer to the question

Reply:
But i have coded according to  you have need, which satisfices all the conditions.